csv2bufr/__init__.py

```diff
@@ -196,258 +168,303 @@ def validate_value(key: str, value: Union[NUMBERS],
+                            value = element["value"]
+                        elif column is not None:
+                            # get column name
+                            # make sure column is in data_dict
+                            if (column not in data_dict):
+                                message = f"column '{column}' not found in data dictionary"  # noqa
+                                raise ValueError(message)
+                            value = data_dict[column]
+                        elif jsonpath is not None:
+                            query = parser.parse(jsonpath).find(metadata)
+                            assert len(query) == 1
+                            value = query[0].value
+                        else:
+                            LOGGER.debug(f"value and column both None for element {element['eccodes_key']}")  # noqa
+                        # ===============================
+                        # apply specified scaling to data
+                        # ===============================
+                        if value in MISSING:
+                            value = None
+                        else:
+                            scale = None
+                            offset = None
+                            if "scale" in element:
+                                scale = element["scale"]
+                            if "offset" in element:
+                                offset = element["offset"]
+                            value = apply_scaling(value, scale, offset)
+                        # ==================
+                        # now validate value
+                        # ==================
+                        valid_min = None
+                        valid_max = None
+                        if "valid_min" in element:
+                            valid_min = element["valid_min"]
+                        if "valid_max" in element:
+                            valid_max = element["valid_min"]
+                        LOGGER.debug(f"validating value {value} for element {element['eccodes_key']}")  # noqa
+                        value = validate_value(element["eccodes_key"], value,
+                                               valid_min, valid_max,
+                                               NULLIFY_INVALID)
+
+                        LOGGER.debug(f"value {value} validated for element {element['eccodes_key']}")  # noqa
+                        # ==================================================
+                        # at this point we should have the eccodes key and a
+                        # validated value to use, add to dict
+                        # ==================================================
+                        self.set_element(eccodes_key, value)
+                        LOGGER.debug(f"value {value} updated for element {element['eccodes_key']}")  # noqa
+            rows_read += 1
+
+
+    def datetime(self):
```

Comment:
rename to `get_datetime`, add function annotation, add docstring

Reply:
done